### PR TITLE
[SPARK-25838][ML] Remove formatVersion from Saveable

### PR DESCRIPTION
**Pull request by mgaido91**
## What changes were proposed in this pull request?

`Saveable` interface introduces `formatVersion` which is protected and it is used nowhere. So the PR proposes to remove it.

## How was this patch tested?

existing tests

### CODE REVIEW

**Comment by viirya** [comment]
cc @mengxr any idea why we have this `formatVersion`?

**Comment by dongjoon-hyun** [comment]
Retest this please.

File: mllib/src/main/scala/org/apache/spark/mllib/classification/LogisticRegression.scala [hunk 1]
```diff
@@ -163,8 +163,6 @@ class LogisticRegressionModel @Since("1.3.0") (
       numFeatures, numClasses, weights, intercept, threshold)
   }
 
-  override protected def formatVersion: String = "1.0"
-
```
Comment (dongjoon-hyun) [inline mllib/src/main/scala/org/apache/spark/mllib/classification/LogisticRegression.scala]: This seems visible to the users. Although this occurs in Spark 3.0.0, I guess we had better do deprecation processes first for this kind of removals.

cc @srowen , @mengxr , @rxin .

**Comment by rxin** [comment]
Who introduced this? We should ask the person that introduced it whether it can be removed.

**Comment by mgaido91** [comment]
this was introduced by @jkbradley in https://github.com/apache/spark/commit/975bcef467b35586e5224171071355409f451d2d.

File: mllib/src/main/scala/org/apache/spark/mllib/classification/LogisticRegression.scala (hunk 1 above)
Comment (mgaido91) [reply mllib/src/main/scala/org/apache/spark/mllib/classification/LogisticRegression.scala]: what do you mean by visible to users? It is protected, it is visible only if a user extends the class...

**Comment by mgaido91** [comment]
retest this please

**Comment by rxin** [comment]
Perhaps @jkbradley and @mengxr can comment on it. If the trait is inheritable, then protected still means it is part of the API contract.

**Comment by mgaido91** [comment]
retest this please

**Comment by srowen** [comment]
Just saw this after reviewing a lot of old PRs. Yes this can be removed. It's old and not used, and think it was superseded by the "SaveLoadVx_0" class pattern. I think it can be removed in Spark 3 as it's not user visible.

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**